components/datetime/src/format/time_zone.rs

```diff
@@ -54,258 +54,6 @@ where
     T: TimeZoneInput,
     W: fmt::Write + ?Sized,
 {
-    match &time_zone_format.kind {
-        TimeZoneFormatKind::Pattern(patterns) => {
-            write_pattern(time_zone_format, time_zone, patterns, w)
-        }
-        TimeZoneFormatKind::Config(config) => write_config(time_zone_format, time_zone, *config, w),
-    }
-}
-
-fn write_config<T, W>(
-    time_zone_format: &TimeZoneFormat,
-    time_zone: &T,
-    config: TimeZoneFormatConfig,
-    w: &mut W,
-) -> Result<(), Error>
-where
-    T: TimeZoneInput,
-    W: fmt::Write + ?Sized,
-{
-    match config {
-        TimeZoneFormatConfig::GenericNonLocationLong => {
-            time_zone_format
-                .long_generic_non_location_format(w, time_zone)
-                .or_else(|_| time_zone_format.generic_location_format(w, time_zone))
```

Comment:
There are 2 types of default fallbacks, one is `generic_location_format->localized_gmt_format` and another is `localized_gmt_format` alone.

In this case, I plan to create an enum to control this behavior.

Reply:
That sounds good to me.

Reply:
reverted the change and we still need it.

Reply:
yeah, that's right. And we can also chain them together. 

Also, there is still so clean up needed after this PR. For example, we don't need to have `TimeZoneFormatConfig` anymore because user can call load methods directly.